components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java

```diff
@@ -57,33 +57,85 @@
 public class GoogleOAuth2Authenticator extends OpenIDConnectAuthenticator {
 
     private static final long serialVersionUID = -4154255583070524018L;
+    private String tokenEndpoint;
+    private String oAuthEndpoint;
+    private String userInfoEndpoint;
 
     private static Log log = LogFactory.getLog(GoogleOAuth2Authenticator.class);
 
     /**
+     * initiate tokenEndpoint
+     */
+    private void initTokenEndpoint() {
+        this.tokenEndpoint = getAuthenticatorConfig().getParameterMap().get(GoogleOAuth2AuthenticationConstant
+                .GOOGLE_TOKEN_ENDPOINT);
+        if (StringUtils.isBlank(this.tokenEndpoint)) {
+            this.tokenEndpoint = IdentityApplicationConstants.GOOGLE_TOKEN_URL;
+        }
+    }
+
+    /**
+     * initiate authorization server endpoint
+     */
+    private void initOAuthEndpoint() {
+        this.oAuthEndpoint = getAuthenticatorConfig().getParameterMap().get(GoogleOAuth2AuthenticationConstant
+                .GOOGLE_OAUTH_ENDPOINT);
+        if (StringUtils.isBlank(this.oAuthEndpoint)) {
+            this.oAuthEndpoint = IdentityApplicationConstants.GOOGLE_OAUTH_URL;
+        }
+    }
+
+    /**
+     * initiate userInfoEndpoint
+     */
+    private void initUserInfoEndPoint() {
+        this.userInfoEndpoint = getAuthenticatorConfig().getParameterMap().get(GoogleOAuth2AuthenticationConstant
+                .GOOGLE_USERINFO_ENDPOINT);
+        if (StringUtils.isBlank(this.userInfoEndpoint)) {
+            this.userInfoEndpoint = IdentityApplicationConstants.GOOGLE_USERINFO_URL;
+        }
+    }
+
+    /**
+     * Get UserInfo Endpoint
+     *
+     * @return userInfoEndpoint
+     */
+
+    private String getUserInfoEndpoint() {
```

Comment:
Shall we pass Map<String, String> authenticatorProperties here as well to be consistent.